### PR TITLE
Feature/user feedback

### PR DESCRIPTION
**Pull request by JanPSchwarz**
Log:

- pages `error.js` & `404.js` added
- `handleError` function added in `_app.js` which routes to `error.js` with the corresponding error message
- `handleError` passed to `fetch` methods to redirect user to `error.js`

- `isLoading` and `error` from initial and central `data fetch` on `_app.js` passed to `Layout.js` to display loading and error globally before rendering rest of the app
- `Loader.js` and `ErrorMessage` components used in `create/slug/[id]` for fetching data on id route

NOTE:
- there are built in random errors in this code on purpose for testing/QA. Feel free to open deployment and try out
   - the errors can appear everytime when using the app as well as on trying to highlight an emotion entry on emotion records page
- the errors will be removed when QA is done of course

### CODE REVIEW

File: components/Loader.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  border-left-color: var(--section-background);
+  border-radius: 50%;
+  width: 40px;
+  height: 40px;
+  animation: spin 1.5s linear infinite;
+`;
+
+//change so if prop fullscreen= true has blurred screen like conf message
+
+export default function Loader({ itemText, fullScreen }) {
```
Comment (SwethaJanardhana) [inline components/Loader.js:26]: I think `fullScreen` prop is not used anymore, and can be removed

File: pages/_app.js [hunk 2]
```diff
@@ -8,10 +8,26 @@ import generateExampleData from "@/utils/exampleData";
...
+    const error = new Error("An error occurred while fetching the data.");
+    // Attach extra info to the error object.
+    error.info = await res.json();
+    error.status = res.status;
+    throw error;
+  }
 
-const fetcher = (url) => fetch(url).then((response) => response.json());
+  return res.json();
+};
```
Comment (SwethaJanardhana) [inline pages/_app.js:27]: `response` instead of shorthand notation `res`

File: pages/api/emotionEntries/index.js [hunk 3]
```diff
@@ -6,6 +6,17 @@ export default async function handler(request, response) {
...
+      // function randomNumber(count) {
+      //   return Math.floor(Math.random() * count);
+      // }
+
+      // const number = randomNumber(2);
+
+      // if (number === 1) {
+      //   throw new Error("Error in fetching emotion Entries");
+      // }
+      ////
```
Comment (SwethaJanardhana) [inline pages/api/emotionEntries/index.js:19]: Commented code can be removed

File: pages/create/[slug]/index.js [hunk 4]
```diff
@@ -21,11 +23,15 @@ export default function EmotionEntry({
   const { id } = router.query;
   const { slug } = router.query;
```
Comment (SwethaJanardhana) [inline pages/create/[slug]/index.js:24]: `const { id, slug } = router.query` will also do

File: pages/index.js [hunk 5]
```diff
@@ -9,13 +9,15 @@ import {
...
+import Loader from "@/components/Loader";
 import dynamic from "next/dynamic";
 import { breakpoints } from "@/utils/breakpoints";
 import Head from "next/head";
 import ToggleSwitch from "@/components/ToggleSwitch";
 
 const TensionChart = dynamic(() => import("../components/TensionChart"), {
   ssr: false,
+  loading: () => <Loader itemText="... loading" />,
 });
```
Comment (SwethaJanardhana) [inline pages/index.js:21]: 👏🏻 performance improvisation

**Review by SwethaJanardhana — COMMENTED** [review]
Very well implementation for error handling, left few hints...

**Comment by JanPSchwarz** [comment]
> Very well implementation for error handling, left few hints...

thank you very much :)

**Pull request merged**